### ISSUE TITLE
W3C compliance of attributes and elements

### ISSUE THREAD

**Issue by MadhukarMallia** [angular/material#3458]
Hi,

I am using Material Angularjs (0.9.0). When I check the W3C compliance of the attributes and elements used, I get an error saying the attributes and elements are not allowed. Tried using data-* attribute with the Material Angularjs attributes (layout and related attributes are being used frequently) but then layout structure breaks. Also, would like to know if this is sorted out in the latest version, so that I can migrate to latest one. Any kind of help is greatly appreciated. 

Thanks in advance.

**Comment by epelc** [comment]
The layout attributes are just css. I think it'd be a lot of bloat to add extra styles for everything with `data-` prefixed.

**Comment by DavidSouther** [comment]
**tl;dr** you cannot use Angular Material and the w3c validator together.

See https://github.com/angular/material/issues/3453#issuecomment-115341734
